packages/iota/examples/mint-native-tokens/src/index.ts

```diff
@@ -0,0 +1,828 @@
+    // We also top up the address by asking funds from the faucet.
+    [ctx.walletAddressHex, ctx.walletAddressBech32, ctx.walletKeyPair] = await setUpHotWallet(ctx.info.protocol.bech32HRP);
+
+    // Fetch outputId with funds to be used as input from the Indexer API
+    const indexerPluginClient = new IndexerPluginClient(ctx.client);
+    // Indexer returns outputIds of matching outputs. We are only interested in the first one coming from the faucet.
+    const outputId = await fetchAndWaitForBasicOutput(ctx.walletAddressBech32, indexerPluginClient);
+
+    console.log("OutputId: ", outputId);
+
+    // Fetch the output itself from the core API
+    const resp = await ctx.client.output(outputId);
+    // We start from one Basic Output that we own, Our journey starts with the genesis.
+    const genesisOutput = resp.output;
+
+    console.log("Genesis output: ", genesisOutput);
+
+    // Prepare a transaction that mints an alias
+    let txPayload1 = mintAliasTx(genesisOutput, outputId, ctx.walletAddressHex, ctx.walletKeyPair, ctx.info);
+    ctx.txList.push(txPayload1);
+
+    // Ready with the alias minting tx, now we have to prepare the second tx that:
+    //   - creates the foundry
+    //   - mints tokens to target address
+    let txPayload2 = createFoundryMintTokenTx(getOutput("tx1Alias"), getOutputId("tx1Alias"), ctx.walletAddressHex, ctx.walletKeyPair, ctx.info, ctx.targetAddress);
+    ctx.txList.push(txPayload2);
+
+    // Tx3 transfers the control of alias to the user address
+    // Tx 3 is going to give the governor role of tha alias to targetaddress
+    let txPayload3 = transferAliasTx(getOutput("tx2Alias"), getOutputId("tx2Alias"), ctx.walletAddressHex, ctx.walletKeyPair, ctx.info, ctx.targetAddress);
+    ctx.txList.push(txPayload3);
+
+    // Finally, time to prepare the three blocks, and chain them together via `parents`
+    let blocks: IBlock[] = await chainTrasactionsViaBlocks(ctx.client, ctx.txList, ctx.info.protocol.minPoWScore);
+
+    // send the blocks to the network
+    // We calculated pow by hand, so we don't define a localPow provider for the client so it doesn't redo the pow again.
+    submit(blocks, ctx.client);
+}
+
+run()
+    .then(() => console.log("Done"))
+    .catch(err => console.error(err));
+
+
+// Transaction creating functions
+
+// Create an alias
+// inputs: a basic output received from faucet
+// outputs: an alias output
+function mintAliasTx(consumedOutput: OutputTypes, consumedOutputId: string, walletAddressHex: string, walletKeyPair: IKeyPair, info: INodeInfo): ITransactionPayload {
+    // Prepare inputs to the tx
+    const input: IUTXOInput = {
+        type: UTXO_INPUT_TYPE,
+        transactionId: consumedOutputId.slice(0, 2 + 2 * TRANSACTION_ID_LENGTH), // +2 because it has 0x prefix
+        transactionOutputIndex: parseInt(consumedOutputId.slice(2 + 2 * TRANSACTION_ID_LENGTH))
```

Comment:
Maybe we can use transactionId and outputIndex from outputResponse metadata, that we retrieve on line 114, instead of slicing consumedOutputId:
`transactionId: resp.metadata.transactionId, transactionOutputIndex: resp.metadata.outputIndex`